overlord/servicestate/servicestate.go

```diff
@@ -127,7 +127,12 @@ func serviceControlTs(st *state.State, appInfos []*snap.AppInfo, inst *Instructi
 		sort.Strings(explicitSvcs)
 		cmd.ExplicitServices = explicitSvcs
 
-		summary := fmt.Sprintf("Run service command %q for services %q of snap %q", cmd.Action, svcs, cmd.SnapName)
+		var summary string
+		if inst.Action == "restart" && len(explicitSvcs) == 0 {
+			summary = fmt.Sprintf("Run service command %q for running services of snap %q", cmd.Action, cmd.SnapName)
+		} else {
+			summary = fmt.Sprintf("Run service command %q for services %q of snap %q", cmd.Action, svcs, cmd.SnapName)
```

Comment:
It would great to have a test for this (also covering stop/start summary for completeness).

Reply:
Thanks, this actually helped find another issue too :-)

Done!